lib/file-dependencies.rb

```diff
@@ -23,8 +23,8 @@ def process_downloads(files, target, tmpdir)
       full_target = file['target'] ? ::File.join(target, file['target']) : target
       download = FileDependencies::File.fetch_file(file['url'], file['sha1'], tmpdir)
       if (res = download.match(/(\S+?)(\.tar\.gz|\.tgz)$/))
-        prefix = res.captures.first.gsub("#{tmpdir}/", '')
         FileDependencies::Archive.untar(download) do |entry|
+          prefix = file['include_tar_prefix'] ? '' : ::File.join(entry.full_name.split("/").first, '')
```

Comment:
Why join with "" ? File.join is platform-agnostic (which is cool!) so maybe we are trying to add a trailing "/" to the prefix? If so, do we need to fix the full_name.split call to use the filesystem separator?